### PR TITLE
Add sendToGithub to Jenkins and wire up into webapp-test job.

### PR DESCRIPTION
**Pull request by jeresig**
## Summary:
This wires up the new Github status check logic added to alertlib into the notify.groovy logic - making it so that the status check is added to the current commit SHA that we're working against. I made sure to add in calls to `sendToGithub` in the `call` method on both build start and completion, this will allow us to see the "pending" state more accurately in Github.

Issue: FEI-4621

## Test plan:
I was able to test this by taking an existing test run:
https://jenkins.khanacademy.org/job/deploy/job/webapp-test/142238/

and then I clicked "Replay" and modified the Main Script and Notify to match what's in this PR and I was able to get working results!
https://jenkins.khanacademy.org/job/deploy/job/webapp-test/142250/

Pending:
Screen Shot 2022-06-23 at 1 18 30 PM

Back to normal:
Screen Shot 2022-06-23 at 1 16 17 PM

Success:
Screen Shot 2022-06-23 at 1 19 00 PM

### CODE REVIEW

**Review by csilvers — APPROVED** [review]
This looks great!  Thanks so much for adding this.

> I have no idea how to safely test this! Any suggestions on how to test this would be greatly appreciated.

You can test this using the "replay" functionality on jenkins.  But you'll have to break this into two parts.

First, have a separate commit that updates alertlib, and land that.  Since no existing funcionality was changed, that should be safe.

Then, once that's been landed, you can visit any job that's safe to run again, e.g.
https://jenkins.khanacademy.org/job/deploy/job/webapp-test/142216/

Then click on replay:
https://jenkins.khanacademy.org/job/deploy/job/webapp-test/142216/replay/

Manually cut-and-paste your new notify.groovy text into the `notify` box on the relpay page.

Hit `Run`.

Profit!

**Comment by csilvers** [comment]
> Manually cut-and-paste your new notify.groovy text into the notify box on the relpay page.

Oh, you'll also need to modify the `Main script` section with your changes that call github.

File: vars/notify.groovy [hunk 1]
```diff
@@ -184,6 +184,44 @@ def sendToSlack(slackOptions, status, extraText='') {
...
+// Supported options:
+//  - sha: The commit SHA to update the status of.
+//  - repo: The repo to update the status in (defaults to webapp).
+//  - owner: The owner of the repo (defaults to Khan).
+def sendToGithub(githubOptions, status, extraText='') {
+   def arr = _dataForAlertlib(status, extraText);
+   def subject = arr[0];
+   def severity = arr[1];
+   def body = arr[2];
+   subject += "${currentBuild.displayName}";
```
[the remark this answers is not in the document]
Comment (jeresig) [reply vars/notify.groovy:197]: I'm not sure! The other `sendTo...` methods seem to do the same thing, which I found surprising! I guess I'll see what happens in testing...

**Comment by jeresig** [comment]
@csilvers Thanks for the testing hints! I'll toss up a separate audit to update `alertlib` and then test this manually.

**Comment by jeresig** [comment]
@csilvers I confirmed it working! (after some bug fixes) Thanks for the testing tips, I've included some proof in the test plan above.

**Review by dbraley — APPROVED** [review]
Cool, nice!

**Pull request merged**